app/models/review.rb

```diff
@@ -2,14 +2,16 @@ class Review < ApplicationRecord
   belongs_to :draft
   belongs_to :user
 
-  validates :user, uniqueness: { scope: :draft,
-                                 message: "You already have a review for this draft" }
+  VALIDATION_MESSAGE = "You already have a review for this draft"
+
+  validates :user_id, uniqueness: { scope: :draft_id,
+                                 message: VALIDATION_MESSAGE }
 
   scope :by_user, ->(user) {
     where(user: user)
   }
 
   def authored_by?(user)
-    !self.nil? && self.user == user
+    !self.user.nil? && self.user == user
```

Comment:
Thank you for taking a look at this.  Looking at this now, I am not 100% sure what the original intent was here.  In the two places we are calling authored_by?(user), user cannot be nil due to controller before_actions that require a current_user before the pages are rendered.  So, I think we can re-write it as:
def authored_by?(potential_author)
user == potential_author
end

Granted, this is not perfect, but I am pretty sure the nil check is extraneous in the contexts we are calling it and I needed to get rid of the local variable 'user' comparison to 'self.user'...that's just confusing 😬

Reply:
@Pchao93 on second thought, I think the nil check needs to stay.  Even though we don't need it at the moment, could definitely lead to an unexpected result in the future.